src/btree/bt_read.c

```diff
@@ -784,8 +784,13 @@ __wt_page_in_func(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags
 			    __evict_force_check(session, ref)) {
 				++force_attempts;
 				ret = __wt_page_release_evict(session, ref, 0);
-				/* If forced eviction fails, stall. */
-				if (ret == EBUSY) {
+				/*
+				 * If forced eviction succeeded, don't retry.
+				 * If it failed, stall.
+				 */
+				if (ret == 0)
+					evict_skip = true;
```

Comment:
I don't think it was causing the current issue, but once we have successfully evicted a page (which might be an update/restore), we shouldn't keep retrying.